### PR TITLE
Respect SETTINGS_HEADER_TABLE_SIZE http2 setting

### LINKED ISSUE #1928: Respect SETTINGS_HEADER_TABLE_SIZE http2 setting
Refs:
Spec: http://httpwg.org/specs/rfc7540.html#SettingValues
hpack Encoder setting: https://godoc.org/golang.org/x/net/http2/hpack#Encoder.SetMaxDynamicTableSizeLimit
hpack Decoder setting: https://godoc.org/golang.org/x/net/http2/hpack#Decoder.SetAllowedMaxDynamicTableSize

Note that the x/net/http2 server doesn't natively support this setting.

### PR DESCRIPTION
**Pull request by lyuxuan**
fix #1928

### CODE REVIEW

**Comment by mumoshu** [comment]
@lyuxuan This worked like a charm! nginx-ingress-controller is now happy to load-balance grpc over h2c workloads to the grpc server w/ this change. Thanks a lot for the fix 👍

File: transport/transport_test.go [hunk 1]
```diff
@@ -2086,3 +2086,113 @@ func runPingPongTest(t *testing.T, msgSize int) {
...
+		if limits.getLen() != 1 {
+			time.Sleep(10 * time.Millisecond)
+			continue
+		}
+		if val := limits.getIndex(0); val != uint32(100) {
+			t.Fatalf("expected limits[0] = 100, got %d", val)
+		}
+		break
+	}
+	if i == 1000 {
```
Comment (MakMukhi) [inline transport/transport_test.go]: Same thing `i` is local to the for loop above.

**Comment by him0** [comment]
I cannot find this PR at release note , but found this changes in version 1.15.0(8dea3dc473e90c8179e519d91302d0597c0ca1d1). Has this pull request already released?
https://github.com/grpc/grpc-go/blob/8dea3dc473e90c8179e519d91302d0597c0ca1d1/internal/transport/controlbuf.go#L31-L33

**Comment by lyuxuan** [comment]
@him0, this feature is released since version 1.13.0. You can look at the commit page [here](https://github.com/grpc/grpc-go/commit/9a54b9abadc29235e480ec8837c5d49747611694), and figure out the versions having this feature. We forgot to include it in the 1.13.0 release note, which is an unfortunate oversight. Thanks!

**Pull request merged**